ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala

```diff
@@ -113,14 +128,20 @@ object Main {
       case _ =>
     }
 
-    val serviceF: Future[HttpService.Error \/ ServerBinding] =
-      HttpService.start(
-        startSettings = config,
-        contractDao = contractDao,
+    val serviceF: Future[HttpService.Error \/ ServerBinding] = {
+      metricsResource.asFuture.flatMap(implicit metrics =>
+        HttpService.start(
+          startSettings = config,
+          contractDao = contractDao,
+        )
       )
+    }
 
     discard {
       sys.addShutdownHook {
+        metricsResource
+          .release()
```

Comment:
In line 149 we call `terminate` which also releases the resource. Are we releasing it twice now?

Reply:
Now we're speaking of it, I think yes.
I'm not particularly fond of the resource freeing here but without different handling of the shutdown hook this needs to stay like it is.

Reply:
(oh what have I done, now I'm here twice :facepalm:)

Reply:
Will address this with a seperate PR